scopes/lanes/lanes/lanes.main.runtime.ts

```diff
@@ -750,8 +753,64 @@ please create a new lane instead, which will include all components of this lane
       )
     );
 
-    const results = await pMapSeries(diffProps, async ({ componentId, sourceHead, targetHead }) =>
-      this.componentDiffStatus(componentId, sourceHead, targetHead, options)
+    const snapDistancesByComponentId = new Map<
+      string,
+      {
+        snapsDistance: SnapsDistance;
+        sourceHead: string;
+        targetHead?: string;
+        componentId: ComponentID;
+      }
+    >();
+
+    await Promise.all(
+      diffProps.map(async ({ componentId, sourceHead, targetHead }) => {
+        const snapsDistance = await this.getSnapsDistance(componentId, sourceHead, targetHead, false);
+        if (snapsDistance) {
+          snapDistancesByComponentId.set(componentId.toString(), {
+            snapsDistance,
+            sourceHead,
+            targetHead,
+            componentId,
+          });
+        }
+      })
+    );
+
+    const commonSnapsToImport = compact(
+      [...snapDistancesByComponentId.values()].map((s) =>
+        !s.snapsDistance.commonSnapBeforeDiverge
+          ? null
+          : `${s.componentId.toStringWithoutVersion()}@${s.snapsDistance.commonSnapBeforeDiverge}`
+      )
+    );
+
+    const sourceOrTargetLane =
+      ((sourceLaneId.isDefault() ? null : (sourceLane as Lane)) ||
+        (targetLaneId?.isDefault() ? null : (targetLane as Lane))) ??
+      undefined;
+
+    if (commonSnapsToImport.length > 0 && !options?.skipChanges) {
+      await this.scope.legacyScope.scopeImporter.importWithoutDeps(
+        ComponentIdList.fromStringArray(commonSnapsToImport),
+        {
+          cache: true,
+          reason: `get the common snap for lane diff`,
+          lane: sourceOrTargetLane,
+        }
+      );
+    }
+
+    const results = await Promise.all(
```

Comment:
if this runs `component.load`, then it can't be in parallel. 
otherwise, I'd be better to limit the parallelism with pmap to 50. (`concurrentComponentsLimit()`)